### PR TITLE
[WIP] add source spaces to viz.plot_bem

### PR DESCRIPTION
**Pull request by christianbrodbeck**
@Eric89GXL testing your suggestion from #3429, plotting all sources in a given slice (an ico-4 source space):

better

### CODE REVIEW

**Comment by Eric89GXL** [comment]
So say roughly 100 slices front to back for the brain, and roughly 8000
source space points means about 80 per slice. Is that what you get? It
looks like more than that.

**Comment by christianbrodbeck** [comment]
oh right I forgot to use n-use :)

**Comment by agramfort** [comment]
Is it slow?

**Comment by christianbrodbeck** [comment]
plot at the top is updated

**Comment by Eric89GXL** [comment]
Seems like it should be about the same speed as plotting the BEM surfaces,
similar vertex counts

**Comment by Eric89GXL** [comment]
Cool plot

File: mne/viz/misc.py [hunk 1]
```diff
@@ -346,6 +365,10 @@ def plot_bem(subject=None, subjects_dir=None, orientation='coronal',
         'coronal' or 'axial' or 'sagittal'.
     slices : list of int
         Slice indices.
+    brain_surfaces : None | str | list of str
+        One or more brain surface to plot (optional). Entries should correspond
+        to files in the subject's ``surf`` directory, e.g., ``"white"`` will
+        plot ``<subject>/surf/lh.white`` and ``<subject>/surf/rh.white``.
```
Comment (agramfort) [inline mne/viz/misc.py]: Update docstring

Comment (christianbrodbeck) [inline mne/viz/misc.py]: This was just proof of concept :) I can also include it in the other plot_bem PR if this plot is satisfactory

Comment (agramfort) [inline mne/viz/misc.py]: would be useful especially for deep structures. Can you check that it works too for a volume or discrete source space?

**Comment by christianbrodbeck** [comment]
> would be useful especially for deep structures. Can you check that it works too for a volume or discrete source space?

```
mne.viz.plot_bem('sample', dp+'/subjects' , brain_surfaces='white', src='volume-7mm-src.fif')
```
vol7

some slices fall in between the grid.... does not look as expected... should there be something like a fallback to look for the closest sources when none fall into a slice? But that might produce unwanted results at other time...

**Comment by agramfort** [comment]
I think this is useful at least for educational purposes. Can you make CIs happy and see where to show this in the doc?

**Comment by christianbrodbeck** [comment]
axial
sagittal

**Comment by Eric89GXL** [comment]
Regarding `Report` I think you mentioned not knowing where this could go, I was thinking the `add_bem_*` function:

https://github.com/mne-tools/mne-python/blob/master/mne/report.py#L997

**Comment by Eric89GXL** [comment]
...and the plots LGTM

**Comment by christianbrodbeck** [comment]
@agramfort Is this tutorial a good place to add it to?

**Comment by christianbrodbeck** [comment]
@Eric89GXL the `Report` does not seem to share any code with `viz.plot_bem` so I suppose it would probably be a larger refactoring project

**Comment by Eric89GXL** [comment]
Ahh you're right, I thought it called it under the hood. That's too bad :(

**Comment by Eric89GXL** [comment]
That looks like a good place to put the examples, please add a link to the CircleCI rendering once it's done to your satisfaction :)

**Comment by jasmainak** [comment]
@christianbrodbeck the report does share some code with `viz.plot_bem`. Please look here: https://github.com/mne-tools/mne-python/blob/master/mne/report.py#L122 and https://github.com/mne-tools/mne-python/blob/aeceebdfc8d4e21d594263fc6b1304d209446944/mne/viz/misc.py#L401. The shared code is in `_plot_mri_contours`

**Comment by christianbrodbeck** [comment]
The rendered report is [here](https://1962-1301584-gh.circle-artifacts.com/0//home/ubuntu/mne-python/doc/_build/html/auto_tutorials/plot_forward.html).

**Comment by Eric89GXL** [comment]
Looks good

**Comment by christianbrodbeck** [comment]
Thanks @jasmainak I missed that! 

So what aspects should be added to the report? I suppose for `.add_bem_to_section()` both `brain_surfaces` and `src` could be exposed. Should the default change? And for `.parse_folder()?` Should we just add the `white` surface by default?

**Comment by christianbrodbeck** [comment]
Also, it would probably be nice if a source space could be added with e.g. `src='ico-4'` (or @Eric89GXL `scr='ico4'` like `setup_source_space()`?

**Comment by agramfort** [comment]
good enough for me. Next in another PR

thanks @christianbrodbeck

**Pull request merged**